library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -231,13 +258,29 @@ libspdm_process_opaque_data_supported_version_data(libspdm_context_t *spdm_conte
         (opaque_element_support_version->version_count == 0)) {
         return LIBSPDM_STATUS_INVALID_MSG_FIELD;
     }
+
+    version_count = opaque_element_support_version->version_count;
+
+    if ((opaque_element_table_header->id != SPDM_REGISTRY_ID_DMTF) ||
+        (opaque_element_table_header->vendor_len != 0) ||
+        (opaque_element_table_header->opaque_element_data_len !=
+         sizeof(secured_message_opaque_element_supported_version_t) +
+         sizeof(spdm_version_number_t) * version_count)) {
+        return LIBSPDM_STATUS_INVALID_MSG_FIELD;
+    }
+
+    if (data_in_size !=
+        libspdm_get_untrusted_opaque_data_supported_version_data_size(spdm_context, version_count)) {
```

Comment:
should it be "<" ?

Reply:
libspdm_get_untrusted_opaque_data_supported_version_data_size works to return 
1. general_opaque_data_table_header_t +
2. secured_message_opaque_element_table_header_t +
3. secured_message_opaque_element_supported_version_t +
4. spdm_version_number_t * version count =
all the opaque data field, right? 
If right, then data_in_size should equals to the return of libspdm_get_untrusted_opaque_data_supported_version_data_size

Reply:
Fine, let's keep "!=" in this patch, because I found more places need change.

Later, we should consider the situation that opaque data include more entries.